/*
 * Copyright 2006 Sun Microsystems, Inc.  All rights reserved.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and associated documentation files (the
 * "Software"), to deal in the Software without restriction, including
 * without limitation the rights to use, copy, modify, merge, publish,
 * distribute, and/or sell copies of the Software, and to permit persons
 * to whom the Software is furnished to do so, provided that the above
 * copyright notice(s) and this permission notice appear in all copies of
 * the Software and that both the above copyright notice(s) and this
 * permission notice appear in supporting documentation.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
 * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
 * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT
 * OF THIRD PARTY RIGHTS. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
 * HOLDERS INCLUDED IN THIS NOTICE BE LIABLE FOR ANY CLAIM, OR ANY SPECIAL
 * INDIRECT OR CONSEQUENTIAL DAMAGES, OR ANY DAMAGES WHATSOEVER RESULTING
 * FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
 * NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION
 * WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 *
 * Except as contained in this notice, the name of a copyright holder
 * shall not be used in advertising or otherwise to promote the sale, use
 * or other dealings in this Software without prior written authorization
 * of the copyright holder.
 *
 */

Fix for: 6541240, P2, Screen saver will lock user out of the system if they have no password set

--- driver/lock.c	Mon Jun 18 12:47:18 2007
+++ driver/lock.c	Fri Jun 15 16:31:12 2007
@@ -1351,6 +1351,9 @@
   if (si->prefs.verbose_p)
     fprintf (stderr, "destroy_passwd_window\n");
 
+  /*reset global flag to indicate passwd dialog is no longer there*/
+  g_passwd_dialog_created = 0;
+
   if (pw == NULL || pw->got_windowid == 0 || si->external_passwd == 0)
     return;
 
@@ -1357,8 +1360,6 @@
   if (pw->timer)
     XtRemoveTimeOut (pw->timer);
 
-  /*reset global flag to indicate passwd dialog is no longer there*/
-  g_passwd_dialog_created = 0;
 
 #ifdef HAVE_XSCREENSAVER_LOCK
   if (si->external_passwd)
